From 5d24cdfbc6937059a0f253da27b2a617a48cc6c9 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mnauw@users.sourceforge.net>
Date: Mon, 22 Feb 2016 13:53:21 +0100
Subject: [PATCH] matroskademux: make stream-id more readable and
 order-friendly

... as streams are so ordered by id by e.g. decodebin
(and as typically already honoured by other demuxers).

(cherry picked from commit afad769c786c02a0cf5c1c3c63a9f4374636e66b)
---
 gst/matroska/matroska-demux.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gst/matroska/matroska-demux.c b/gst/matroska/matroska-demux.c
index 00b65f7..ef48663 100644
--- a/gst/matroska/matroska-demux.c
+++ b/gst/matroska/matroska-demux.c
@@ -1267,7 +1267,8 @@ gst_matroska_demux_add_stream (GstMatroskaDemux * demux, GstEbmlRead * ebml)
 
   stream_id =
       gst_pad_create_stream_id_printf (context->pad, GST_ELEMENT_CAST (demux),
-      "%03" G_GUINT64_FORMAT, context->uid);
+      "%03" G_GUINT64_FORMAT ":%03" G_GUINT64_FORMAT,
+      context->num, context->uid);
   stream_start =
       gst_pad_get_sticky_event (demux->common.sinkpad, GST_EVENT_STREAM_START,
       0);
